docs/02-app/02-api-reference/02-file-conventions/error.mdx

```diff
@@ -71,43 +77,83 @@ export default function Error({ error, reset }) {
 }
 ```
 
+## How `error.js` Works
+
+`error.js` wraps a route segment and its nested children in a [React Error Boundary](https://react.dev/reference/react/Component#catching-rendering-errors-with-an-error-boundary). When an error throws within the boundary, the `error` component shows as the fallback UI.
+
+<Image
+  alt="How error.js works"
+  srcLight="/docs/light/error-overview.png"
+  srcDark="/docs/dark/error-overview.png"
+  width="1600"
+  height="903"
+/>
+
+> **Good to know**:
+>
+> - The [React DevTools](https://react.dev/learn/react-developer-tools) allow you to toggle error boundaries to test error states.
+
 ## Props
 
 ### `error`
 
 An instance of an [`Error`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Error) object forwarded to the `error.js` Client Component.
 
-#### `error.message`
+> **Good to know:** During development, the `Error` object forwarded to the client will be serialized and include the `message` of the original error for easier debugging. However, **this behavior is different in production** to avoid leaking potentially sensitive details included in the error to the client.
 
-The error message.
+#### `error.message`
 
-- For errors forwarded from Client Components, this will be the original Error's message.
-- For errors forwarded from Server Components, this will be a generic error message to avoid leaking sensitive details. `errors.digest` can be used to match the corresponding error in server-side logs.
+- Errors forwarded from Client Components show the original `Error` message.
+- Errors forwarded from Server Components show a generic message with an identifier. This is to prevent leaking sensitive details. You can use the identifier, under `errors.digest`, to match the corresponding server-side logs.
```

Comment:
Do we ever explain how you'd match up client and server errors with `errors.digest`?

Reply:
Good suggestion, maybe we can include more details here.